### PR TITLE
Fixed blank layer for OpenVINO 2022.1

### PR DESCRIPTION
**Pull request by alexlyulkov**
Changed blank layer because it didn't work with old OpenVINO versions(2022.1). The blank layer was implemented using ConvertLike layer, now it is implemented using ShapeOf and Reshape layers.

### Pull Request Readiness Checklist

See details at https://github.com/opencv/opencv/wiki/How_to_contribute#making-a-good-pull-request

- [x] I agree to contribute to the project under Apache 2 License.
- [x] To the best of my knowledge, the proposed patch is not based on a code under GPL or another license that is incompatible with OpenCV
- [x] The PR is proposed to the proper branch
- [x] There is a reference to the original bug report and related work
- [x] There is accuracy test, performance test and test data in opencv_extra repository, if applicable
      Patch to opencv_extra has the same branch name.
- [x] The feature is well documented and sample code can be built with the project CMake

### CODE REVIEW

**Review by asmorkalov — COMMENTED** [review]
:+1:

**Review by asmorkalov — APPROVED** [review]

**Comment by asmorkalov** [comment]
No regressions with 2024.0.0.14509.34caeefd078_x86_64. Also tested with 2022.1.1.7030.39aba80957e. The patch improves pass-rate dramatically. Thanks!

**Comment by asmorkalov** [comment]
@dkurt what do you think?

File: modules/dnn/src/layers/blank_layer.cpp [hunk 1]
```diff
@@ -161,7 +161,11 @@ class BlankLayerImpl CV_FINAL : public BlankLayer
                                         const std::vector<Ptr<BackendNode> >& nodes) CV_OVERRIDE
     {
         auto ieInpNode = nodes[0].dynamicCast<InfEngineNgraphNode>()->node;
-        auto blank = std::make_shared<ov::op::v1::ConvertLike>(ieInpNode, ieInpNode);
+        // We use Reshape as blank layer.
+        // Previously, we used Concat layer, but it didn't work with scalar tensor.
+        // Also we used ConvertLike layer, but it did't work with old OpenVINO versions.
+        auto shape = std::make_shared<ov::op::v0::ShapeOf>(ieInpNode);
+        auto blank = std::make_shared<ov::op::v1::Reshape>(ieInpNode, shape, false);
         return Ptr<BackendNode>(new InfEngineNgraphNode(blank));
```
Comment (opencv-alalek) [inline modules/dnn/src/layers/blank_layer.cpp:169]: May be just return `ieInpNode` without any "operations".

Comment (dkurt) [reply modules/dnn/src/layers/blank_layer.cpp:169]: Unfortunately, that won't work and workaround is to create a dummy node. I've checked that nGraph usually removes such kind of nodes as a part of middle graph processing (like identical Reshape, ConvertLike to the same type and so on).

**Comment by dkurt** [comment]
Shall we target 4.x branch?

**Comment by asmorkalov** [comment]
May be yes, but the issue is not visible with 4.x tests.

**Comment by asmorkalov** [comment]
@alexlyulkov what do you think about 4.x branch?

**Comment by asmorkalov** [comment]
I checked git history and the issue is not relevant for 4.x. It uses slightly different code for the layer initialization.

**Comment by asmorkalov** [comment]
@dkurt Could you take a look?

**Pull request merged**